radix-engine/src/blueprints/resource/fungible/fungible_proof.rs

```diff
@@ -135,18 +137,24 @@ impl FungibleProofBlueprint {
 
     pub(crate) fn drop<Y>(proof: Proof, api: &mut Y) -> Result<(), RuntimeError>
     where
-        Y: ClientApi<RuntimeError>,
+        Y: KernelSubstateApi<SystemLockData> + ClientApi<RuntimeError>,
     {
-        let parent = api
-            .get_object_info(proof.0.as_node_id())?
-            .outer_object
-            .unwrap();
-
-        api.call_method(
-            parent.as_node_id(),
-            RESOURCE_MANAGER_DROP_PROOF_IDENT,
-            scrypto_encode(&ResourceManagerDropProofInput { proof }).unwrap(),
+        // TODO: add `drop` callback for drop atomicity, which will remove the necessity of kernel api.
+
+        // Notify underlying buckets/vaults
+        let handle = api.kernel_lock_substate(
```

Comment:
Similarly here you could just use the substates returned from `drop_object`

Reply:
In the long run, I'd like us to consider removing return of `drop_object`, since it has no semantic meaning (but pure bytes).

Like in Rust, if a struct is destructed, we don't have access to its internal.